baremetal/metal3data_manager.go

```diff
@@ -1269,3 +1277,11 @@ func fetchM3IPClaim(ctx context.Context, cl client.Client, mLog logr.Logger,
 	}
 	return metal3IPClaim, nil
 }
+
+// BuildProviderIDToNodes generates providerID and providerLabel by combining bmhuid and m3muid.
+func BuildProviderIDToNodes(bmhuid string, m3muid string) string {
```

Comment:
I would really appreciate if we could follow [naming conventions for functions](https://www.golangprograms.com/naming-conventions-for-golang-functions.html), since this function is only used within the same package and not exported to other packages, we could name it, i.e  `buildProviderIDToNodes`

Reply:
The function will be used from the controller package in the main PR
https://github.com/metal3-io/cluster-api-provider-metal3/pull/324/files#diff-250f2503d81624c7304619762d56ee502ff1462e5ad0c965af1eb9bfed5f340cR249-R250